docs/adr/0016-auth-for-settings-redesign.md

```diff
@@ -0,0 +1,114 @@
+# Authentication in Settings Redesign
+
+- Status: proposed
+- Deciders: Les Orchard, Lauren Zugai, Ben Bangert, Ryan Kelly
+- Date: 2020-04-15
+
+Issue: https://jira.mozilla.com/browse/FXA-1559
+
+## Context and Problem Statement
+
+The Settings Redesign app needs to query & mutate the same protected user data as the existing settings app hosted on content-server. This will require some form of authentication & authorization to manage that data. [ADR-0014] included several assumptions that turned out to be inaccurate during implementation efforts. Using OAuth2 tokens in the auth-server for the API calls needed is a larger effort than anticipated and will not meet time/work constraints for the Settings Redesign. This ADR includes a new option that is proposed to meet these decision drivers.
```

Comment:
> [ADR-0014] included several assumptions that turned out to be inaccurate during implementation efforts. 

Which assumptions? [Is it mainly this stuff about the levels of auth?](https://github.com/mozilla/fxa/pull/4931#discussion_r411828476) If so, would be helpful to link to that for context in how this decision was made for later.